caramel/compiler/ocaml_to_erlang/fun.ml

```diff
@@ -180,14 +180,20 @@ and mk_expression exp ~var_names ~modules ~functions ~module_name =
               ~f:(Name.atom (Atom.lowercase n_name))
       in
 
+      let rec compute_arity next_part counter =
+        match next_part with
+        | Tarrow (_, _, { desc; _ }, _) -> compute_arity desc (counter + 1)
```

Comment:
I think we will need to do the same follow-trick we did in the uncurry module here, since some types will actually be _pointers_ to other types and we need to dereference them. 

https://github.com/AbstractMachinesLab/caramel/blob/main/caramel/compiler/ocaml_to_erlang/uncurry.ml#L14